src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/ExceptionHandling.cs

```diff
@@ -223,6 +223,26 @@ private static void OnFirstChanceExceptionViaClassLib(object exception)
             }
         }
 
+        private static void OnUnhandledExceptionViaClassLib(object exception)
+        {
+            IntPtr pOnUnhandledExceptionFunction =
+                (IntPtr)InternalCalls.RhpGetClasslibFunctionFromEEType((IntPtr)exception.EEType, ClassLibFunctionId.OnUnhandledException);
+
+            if (pOnUnhandledExceptionFunction == IntPtr.Zero)
+            {
+                return;
+            }
+
+            try
+            {
+                ((delegate*<object, void>)pOnUnhandledExceptionFunction)(exception);
+            }
+            catch when (true)
```

Comment:
```suggestion
            catch
```

Reply:
`catch when (true)` is used in the rest of the file, so I think it is fine to use here as well for consistency.

We use to have a mode where the core runtime was a different type universe from the app (`System.Object` here was not the apps `System.Object`). This is left-over from that mode. `catch when (true)` was needed here to catch all exceptions. Regular `catch` would only catch the exceptions from this type universe.